### ISSUE TITLE
deletes comment from an otherwise empty function body

### ISSUE THREAD

**Issue by tqwhite** [jlongster/prettier#540]
When a comment is the only thing in a function, it formats the code but does not keep the comment. I found this in the course of documenting the previous issue 'assertion errors associated with ...'. The video is the same one. The part specific to this issue is in the last thirty seconds.

[http://media.genericwhite.com/video/?prettierIssue013117Number2](http://media.genericwhite.com/video/?prettierIssue013117Number2)

[prettierIssue013117Number2.zip](https://github.com/jlongster/prettier/files/743411/prettierIssue013117Number2.zip)

**Comment by vjeux** [comment]
If you could provide examples using the web version of prettier that would be awesome: https://jlongster.github.io/prettier/

This bug is a good template to follow: https://github.com/jlongster/prettier/issues/527

**Comment by vjeux** [comment]
As mentioned in the other issue, it seems like this has been fixed on 0.14.1 and you are on an older version. Do you mind updating to the latest version and see if it's still an issue for you? Thanks a lot for reporting this bugs, I can't wait to see more videos from you!

```js
this.persistenceFunction(`error: ${this.loggingIdString} ${response.statusCode}`, '', (err, result) => {
  // perhaps this tells the event queue to keep the event since it's has not been saved successfully
})
```

outputs

```js
this.persistenceFunction(
  `error: ${this.loggingIdString} ${response.statusCode}`,
  "",
  (err, result) => {
    // perhaps this tells the event queue to keep the event since it's has not been saved successfully
  }
);
```

https://jlongster.github.io/prettier/#%7B%22content%22%3A%22this.persistenceFunction(%60error%3A%20%24%7Bthis.loggingIdString%7D%20%24%7Bresponse.statusCode%7D%60%2C%20''%2C%20(err%2C%20result)%20%3D%3E%20%7B%5Cn%20%20%2F%2F%20perhaps%20this%20tells%20the%20event%20queue%20to%20keep%20the%20event%20since%20it's%20has%20not%20been%20saved%20successfully%5Cn%7D)%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

**Comment by tqwhite** [comment]
I forgot to look at the version number before I upgraded but, the newer version FIXED the problem. Sorry I didn't think of updating before I filed the issue. I'll try to remember that in the future. Thanks for your prompt and kind attention.
